jsonschema/validators.py

```diff
@@ -595,9 +598,38 @@ def from_schema(
 
         return cls(base_uri=id_of(schema), referrer=schema, *args, **kwargs)
 
+    def http_handler(self, uri):
+        try:
+            import requests
+        except ImportError:
+            pass
+        else:
+            if hasattr(requests.Response, "json"):
+                session = requests.Session()
```

Comment:
This code essentially creates and discards a session each time it's invoked. Seems like it definitely should be holding onto it instead (and likely allowing a user to pass one in). The previous code would be using the global requests session.

Reply:
Agreed, lazy on my part.

Reply:
Hmm, this is already done internally with `get` by default, so this is no slower than the current approach.